### PR TITLE
Feature/pp9 10573 dynamic output formats

### PR DESCRIPTION
**Pull request by rjs-picturepark**

### CODE REVIEW

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.bottom-sheet {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  height: 50px;
+  background-color: #bb3f30;
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.scss:6]: hardcoded colour

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  div {
+    flex: 1;
+    color: white;
+  }
+}
+
+button {
+  margin-left: 5px;
+  color: black !important;
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.scss:17]: Should these hardcoded colours be replaced by vars?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export class SnackbarComponent implements OnInit {
+  constructor(
+    private snackbarRef: MatBottomSheetRef<SnackbarComponent>,
+    @Inject(MAT_BOTTOM_SHEET_DATA) public data: any
+  ) {}
+
+  title: string;
+  clearBar(): void {
+    this.snackbarRef.dismiss();
+    // event.preventDefault();
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.ts:18]: remove this

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  changeStatus() {
+    this.snackbarRef.dismiss({
+      message: 'Change Status',
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.ts:23]: Translate message

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      data: this.data,
+    });
+  }
+
+  ngOnInit() {
+    console.log('data received', this.data);
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.ts:29]: should this be here?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts [hunk 6]
```diff
@@ -415,6 +415,18 @@ export let TRANSLATIONS = {
     NoteDynamic: {
       en: 'Note: For downloading dynamic formats, please select a single item only.',
     },
+    ConfirmDownloadTitle: {
+      en: 'Download ready',
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts:419]: Add German translations

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-browser/components/content-image-preview/content-image-preview.component.ts [hunk 7]
```diff
@@ -158,16 +158,24 @@ export class ContentImagePreviewComponent extends BaseComponent implements OnCha
...
             contentId: this.content.id,
             outputFormatId: previewOutput.outputFormatId,
           }),
         ],
+        notifyProgress: true,
       });
-
-      const response = await this.contentService.createDownloadLink(request).toPromise();
+      let downloadUrl = '';
+      const DownloadLinkBusinessProcess = await this.contentService.createDownloadLink(downloadLinkRequest).toPromise();
```
[the remark this answers is not in the document]
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-browser/components/content-image-preview/content-image-preview.component.ts:171]: Changes to not capitalized

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 8]
```diff
@@ -56,6 +57,7 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
   public noOutputs = false;
   public hasDynamicOutputs = false;
   public singleItem = false;
+  public waintingDownload = false;
```
[the remark this answers is not in the document]
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:60]: Yes, typo

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 9]
```diff
@@ -165,7 +214,15 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
     const outputs = this.selection.getSelectedOutputs();
     this.hasDynamicOutputs = outputs.some((i) => i.dynamicRendering && !i.detail!.fileSizeInBytes);
     if (outputs.length > 0) {
-      this.fileSize = outputs.map((i) => i.detail!.fileSizeInBytes || 0).reduce((total, value) => total + value);
+      this.fileSize = outputs
+        .map((i) => {
+          if (i.detail) {
+            return i.detail!.fileSizeInBytes || 0;
```
[the remark this answers is not in the document]
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:220]: Changed to 
"return i.detail?.fileSizeInBytes || i.fileSize || 0;"

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 10]
```diff
@@ -226,25 +279,21 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
...
-    });
-    this.subscription.add(outputSubscription);
+    if (this.data.contents.length <= 1000) {
+      const request = new OutputResolveManyRequest({ contentIds: this.data.contents.map((i) => i.id) });
+      this.sub = this.contentService.getOutputsMany(request).subscribe(async (outputs) => {
+        await this.getSelection(outputs, this.data.contents);
+        this.update();
+        this.loader = false;
+      });
+    } else {
```
[the remark this answers is not in the document]
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:296]: Was missing the showing of warning when more then 1000 contents selected

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.html [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<div class="bottom-sheet">
+  <div>Down</div>
```
[the remark this answers is not in the document]
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/snackbar/snackbar.component.html:2]: Forgot to deleted unused component

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.html [hunk 12]
```diff
@@ -6,49 +6,54 @@ <h3>
...
-                  [(ngModel)]="output.selected"
-                  (change)="update()"
-                >
-                  {{ output.name }} ({{ output.values.length }})
-                </mat-checkbox>
-              </div>
-            </div>
-            <div *ngIf="data.mode === 'single'">
-              <mat-radio-group>
+      <ng-container *ngIf="!tooManyContents; else contensMessage">
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.html:9]: else **contensMessage**"> - English/Portuguese mixup?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts [hunk 13]
```diff
@@ -427,6 +427,9 @@ export let TRANSLATIONS = {
     DownloadError: {
       en: 'There was a problem with the download',
     },
+    TooManyContentsError: {
+      en: 'The download of more than 1000 contents is not supported.',
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts:431]: Debatable but, in my opinion, **contents** doesn't fit in the phrase. For example, **items** would be more suitable

Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts:431]: agreed

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.html [hunk 14]
```diff
@@ -5,45 +5,54 @@ <h3>
...
-                <mat-radio-button
-                  *ngIf="!output.hidden && output.values.length"
-                  [value]="filetype.id + '-' + output.id"
-                  [checked]="output.selected"
-                  (change)="radioChange(output, filetype)"
-                >
-                  {{ output.name }} ({{ output.values.length }})
-                </mat-radio-button>
+    <div class="content-wrapper">
+      <ng-container *ngIf="!tooManyContents; else tooManycontentsMessage">
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.html:9]: tooMany"C"ontents

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 15]
```diff
@@ -69,12 +70,15 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
 
   constructor(
     @Inject(MAT_DIALOG_DATA) public data: ContentDownloadDialogOptions,
+    @Inject(PICTUREPARK_CONFIGURATION) private pictureparkConfiguration: PictureparkConfiguration,
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:73]: Check if still needed

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 16]
```diff
@@ -138,13 +141,57 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
 
     const request = new ContentDownloadLinkCreateRequest({
       contents: data.map((i) => ({ contentId: i.contentId, outputFormatId: i.outputFormatId })),
+      notifyProgress: true,
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:144]: Check if can be set to false

Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:144]: Set to false

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 17]
```diff
@@ -138,13 +141,57 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
...
-        window.location.replace(download.downloadUrl);
-        this.dialogRef.close(true);
+    this.waitingDownload = true;
+    this.sub = this.contentService.createDownloadLink(request).subscribe((businessProcess) => {
+      let isTimmerRunning = true;
+      const downloadTimmer = setTimeout(() => {
+        isTimmerRunning = false;
+        this.dialogRef.close();
+        this.snackBar.open(this.translationService.translate('ContentDownloadDialog.DownloadPending'), undefined, {
+          duration: 3000,
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:153]: Increase the duration for 6000

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 18]
```diff
@@ -138,13 +141,57 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
...
+        });
+      }, 8000);
+
+      if (!businessProcess.referenceId) {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:157]: Move this into the subscriber

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 19]
```diff
@@ -138,13 +141,57 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
...
       contents: data.map((i) => ({ contentId: i.contentId, outputFormatId: i.outputFormatId })),
+      notifyProgress: true,
     });
-    const linkSubscription = this.contentService.createDownloadLink(request).subscribe((download) => {
-      linkSubscription.unsubscribe();
-      if (download.downloadUrl) {
-        window.location.replace(download.downloadUrl);
-        this.dialogRef.close(true);
+    this.waitingDownload = true;
+    this.sub = this.contentService.createDownloadLink(request).subscribe((businessProcess) => {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:147]: Create a download facade to join both logics

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts [hunk 20]
```diff
@@ -226,25 +275,16 @@ export class ContentDownloadDialogComponent extends DialogBaseComponent implemen
...
-        contentIds: this.data.contents.map((i) => i.id),
-        renderingStates: [OutputRenderingState.Completed],
-        limit: 1000,
-      })
-    ).subscribe(async (outputs) => {
-      await this.getSelection(outputs.results, this.data.contents);
-      this.update();
-      this.loader = false;
+    const request = new OutputResolveManyRequest({ contentIds: this.data.contents.map((i) => i.id) });
+    this.sub = this.contentService.getOutputsMany(request).subscribe((outputs) => {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:286]: Check if there is a limit for the amount of items

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts [hunk 21]
```diff
@@ -412,8 +412,25 @@ export let TRANSLATIONS = {
     NoAccess: {
       en: 'No access to any format',
     },
-    NoteDynamic: {
-      en: 'Note: For downloading dynamic formats, please select a single item only.',
+    ConfirmDownloadTitle: {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/utilities/translations.ts:415]: Check translations with Nicole

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts (hunk 15 above)
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:73]: No longer needed

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts (hunk 20 above)
Comment (rjs-picturepark) [reply src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-download-dialog/content-download-dialog.component.ts:286]: There isn't

**Pull request merged**